data_source.go

```diff
@@ -355,9 +355,16 @@ func (ds *AnySource) PrepareRun() error {
 	}
 	// Use defaultTS for any channels not in the stored state.
 	// This will be needed any time you have more channels than in the
-	// last saved configuration.
-	defaultTS := TriggerState{AutoDelay: 250 * time.Millisecond,
-		LevelLevel: 4000, EdgeLevel: 100, EdgeRising: true}
+	// last saved configuration. All trigger types are disabled.
+	defaultTS := TriggerState{
+		AutoTrigger:  false,
+		AutoDelay:    250 * time.Millisecond,
+		EdgeTrigger:  false,
+		EdgeLevel:    100,
+		EdgeRising:   true,
+		LevelTrigger: false,
+		LevelLevel:   4000,
+	}
```

Comment:
I think this line should read `defaultTS := TriggetState{}`, I don't see why we should have any default nonzero values. What effect does this have? I guess it effectively sets the text in the dastard commander trigger ui?

It's good enough that I'll stop complaining about it and let you decide.

Reply:
Right: These other values set the text that you'll see in the GUI when you start to use it. They ensure that no nonsensical values appear there.